app/lib/api_support/params.rb

```diff
@@ -6,32 +6,85 @@ module ApiSupport::Params
   included do
     wrap_parameters format: %i[url_encoded_form json multipart_form]
     before_action :wrap_parameters_check
+    before_action :unpermitted_parameters_check
+    class_attribute :_class_action_on_unpermitted_parameters
+    class_attribute :_class_always_permitted_parameters, default: []
```

Comment:
Looks to me that this configuration logic looks mature enough to deserve its own separate module.
Maybe a `ApiSupport::PermittedParams` or `ApiSupport::PermittedParamsConfig`.
We could even include it only in `Admin::Api::BaseController` for now, leaving `Master::Api::BaseController` unchanged.
What do you think?

Reply:
I'd prefer to use a separate module, yes. In fact, I tried, but I found that I needed to call `flat_params`: look at the `unpermitted_keys` method. Besides, `unpermitted_keys` also calls the new method `wrapped_keys` which makes sense when the controller is calling `wrap_parameters`, but not all controllers do that and I don't know if that could cause problems.

Therefore, this module `ApiSupport::Params` is a strong dependency and I probably would need to include it from the new module in the case I create it. I could do it if you think it's a good idea.

One way or another, the logic for both modules will be included on any controller that includes extra params restriction.

Reply:
> I could do it if you think it's a good idea.

I do not 😅